pkg/client/cache/index.go

```diff
@@ -28,6 +28,8 @@ type Indexer interface {
 	Store
 	// Retrieve list of objects that match on the named indexing function
 	Index(indexName string, obj interface{}) ([]interface{}, error)
+	// Retrieve list of all index names
+	ListIndexNames(name string) []string
```

Comment:
Sorry, I think I got confused here.  Is there a test case that you can author to show your desired use of this new function more clearly?  I would have thought you are just looking to get the keys on the ```Indices``` object so name is not needed.  Let me know if I misunderstood something here, but this change seems wrong.